frontend/src/features/recommendation/SearchFilter.tsx

```diff
@@ -25,29 +26,42 @@ function SearchFilter() {
     setExpanded(!expanded);
   };
 
+  const isFilterActive = () => {
+    for (const pair of filterParams.entries()) {
+      if (pair[0] in recommendationFilters && parseInt(pair[1], 10) !== 50) {
+        return true;
+      }
+    }
+    return false;
+  };
```

Comment:
This implementation makes the assumption that any parameter with value `"50"` is necessarily "inactive".  
It could cause subtle bugs when additional filter parameters are introduced. 

What about inverting this loop logic, and check if any recommendation filter has a non-default value, by defining their default values in the constants:
```suggestion
  const isFilterActive = () =>
    Object.entries(defaultRecommendationFilters).some(
      ([key, defaultValue]) =>
        ![null, defaultValue].includes(filterParams.get(key))
    );
```